synapse/lib/stormlib/gen.py

```diff
@@ -425,6 +435,36 @@ class LibGen(s_stormtypes.Lib):
             [ geo:place=(gen, name, $geoname) :name=$geoname ]
             return($node)
         }
+
+        function __getCryptoNodeBySha256(sha256) {
+            // Try to lift crypto:x509:cert by sha256
+            crypto:x509:cert:sha256=$sha256
+            return($node)
+
+            // Try to lift crypto:x509:cert by file
+            crypto:x509:cert:file = { file:bytes:sha256=$sha256 }
+            { -:sha256 [ :sha256 = $sha256 ] }
+            return($node)
+
+            // Create a new crypto:x509:cert with file and sha256
+            [ crypto:x509:cert=(gen, sha256, $sha256,)
+                :file = {[ file:bytes=$sha256 ]}
+                :sha256 ?= $sha256
```

Comment:
If this sha256 is user-supplied then we should do a trycast at the top and return null if it fails.  We would want to cast it before using it in the guid anyway in that case.

Reply:
Actually since this is private and you're already casting in the caller, can probably just drop the tryoper then.